batch/batch/driver/job.py

```diff
@@ -465,15 +465,18 @@ async def schedule_job(app, record, instance):
 
         log.info(f'schedule job {id} on {instance}: called create job')
 
-        rv = await db.execute_and_fetchone(
-            '''
-CALL schedule_job(%s, %s, %s, %s);
-''',
-            (batch_id, job_id, attempt_id, instance.name),
-            'schedule_job',
-        )
+        try:
+            rv = await db.execute_and_fetchone(
+                '''
+    CALL schedule_job(%s, %s, %s, %s);
+    ''',
+                (batch_id, job_id, attempt_id, instance.name),
+                'schedule_job',
+            )
+        except Exception:
+            log.exception(f'error while scheduling job {id} on {instance}')
+            raise
     except Exception:
-        log.exception(f'error while scheduling job {id} on {instance}')
```

Comment:
this would muffle non client response errors from the `post`.

Two thoughts:
1. the adjustment of free cores in memory should happen locally to where the free cores are incremented.
   1. The job private instances never decrement free cores anyway, so this isn't needed
   2. The pool should use a try-except to adjust the free cores (without logging) in `schedule_loop_body` right where the free cores are adjusted
2. Now this outer try-except is irrelevant. Push logging into inner try-excepts if needed (e.g. exactly as you've done with the SQL code)